### PR TITLE
Fix nested module example

### LINKED ISSUE #20609: Nested child LightningModules in docs
### 📚 Documentation

Hi, I've come across this issue where it's explained how a LightningModule should be at the top level and nesting them isn't supported. https://github.com/Lightning-AI/pytorch-lightning/issues/20053#issuecomment-2211752670

However, in the transfer learning docs, I can see this nesting being done. 
https://github.com/Lightning-AI/pytorch-lightning/blob/1f5add327fd88fe288a2f889d720e5d5e06bd7d2/docs/source-pytorch/advanced/transfer_learning.rst

Is this nesting well defined now? How is `CIFAR10Classifier` supposed to behave if `AutoEncoder` uses `self.log()` or has a hook like `predict_step()` defined?

Thanks!

cc @lantiga @borda

### PR DESCRIPTION
**Pull request by matsumotosan**
## What does this PR do?

Fixes #20609

- nested lightning modules are not supported but example had nested modules
- only encoder (`nn.Module`) is needed for feature extractor anyway

<details>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--20923.org.readthedocs.build/en/20923/

### CODE REVIEW

**Comment by Borda** [comment]
@lantiga mind have look, pls 🙏

**Review by Borda — APPROVED** [review]
